low-level/files/opt/seagate/sspl/setup/sspl_config.py

```diff
@@ -0,0 +1,336 @@
+
+        # For node replacement scenario consul will not be running on the new node. But,
+        # there will be two instance of consul running on healthy node. When new node is configured
+        # consul will be brought back on it. We are using VIP to connect to consul. So, if consul
+        # is not running on new node, we dont need to error out.
+        # If consul is not running, exit
+        # Onward LDR_R2, consul will be abstracted out and it won't
+        # exit as hard dependeny of SSPL
+
+        if consts.PRODUCT_NAME == 'LDR_R1':
+            if not os.path.exists(consts.REPLACEMENT_NODE_ENV_VAR_FILE):
+                CONSUL_PS = None
+                for proc in psutil.process_iter():
+                    if 'consul' in proc.name():
+                        CONSUL_PS = proc.pid
+                if not CONSUL_PS:
+                    sys.stderr.write("Consul is not running, exiting..")
+                    sys.exit(1)
+
+        # Get the types of server and storage we are currently running on and
+        # enable/disable sensor groups in the conf file accordingly.
+        if read_provisioner_config:
+            # subprocess.call(['python3', f'{self._script_dir}/conf_based_sensors_enable'])   
+            update_sensor_info()    
+            
+    def get_rabbitmq_cluster_nodes(self):
+        pout = None
+        if self.rabbitmq_major_release == '3' and self.rabbitmq_maintenance_release == '8':
+            rabbitmq_cluster_status = SSPLSetup._send_command('/usr/sbin/rabbitmqctl cluster_status --formatter json')
+            rabbitmq_cluster_status = json.loads(rabbitmq_cluster_status)
+            running_nodes = rabbitmq_cluster_status['running_nodes']
+            for i, node in enumerate(running_nodes):
+                running_nodes[i] = node.replace('rabbit@', '')
+            pout = " ".join(running_nodes)
+        elif self.rabbitmq_version == '3.3.5':
+            out = SSPLSetup._send_command("rabbitmqctl cluster_status | grep running_nodes | cut -d '[' -f2 | cut -d ']' -f1 | sed 's/rabbit@//g' | sed 's/,/, /g'")
+            # pout = SSPLSetup._send_command(f'echo {out} | sed  "s/\'//g" | sed  "s/ //g"')
+            out = out.replace("'", '')
+            pout = out.replace(' ', '')
+        else:
+            sys.stderr.write(f"This RabbitMQ version: {self.rabbitmq_version} is not supported")
+            sys.exit(1)
+        return pout
+
+    def get_cluster_running_nodes(self, msg_broker : str):
+        if(msg_broker == 'rabbitmq'):
+            return self.get_rabbitmq_cluster_nodes()
+        else:
+            self.usage()       
+            
+
+    def process(self):
+        cmd = "config"
+        if(cmd == "config"):
+            self.config_sspl()
+        else:
+            self.usage()
+
+        # Get the version. Output can be 3.3.5 or 3.8.9 or in this format
+        self.rabbitmq_version = SSPLSetup._send_command("rpm -qi rabbitmq-server | awk -F': ' '/Version/ {print $2}")
```

Comment:
currently this cmd is not working as SimpleProcess().run() splits the cmd with spaces,  so looking into it..